src/core/ui/mainui.hpp

```diff
@@ -28,6 +28,9 @@ namespace otto::ui {
   /// Check if a key is currently pressed.
   bool is_pressed(Key k) noexcept;
 
+  using key_handler = std::function<void(Key k)>;
```

Comment:
should probably be `KeyHandler`
only member aliases that are STL-like should be `snake_case`